x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/amazonbedrock/AmazonBedrockInferenceClientCache.java

```diff
@@ -35,20 +35,26 @@ public AmazonBedrockInferenceClientCache(BiFunction<AmazonBedrockModel, TimeValu
     }
 
     public AmazonBedrockBaseClient getOrCreateClient(AmazonBedrockModel model, @Nullable TimeValue timeout) {
-        var returnClient = internalGetOrCreateClient(model, timeout);
         flushExpiredClients();
-        return returnClient;
+        return internalGetOrCreateClient(model, timeout);
     }
 
     private AmazonBedrockBaseClient internalGetOrCreateClient(AmazonBedrockModel model, @Nullable TimeValue timeout) {
         final Integer modelHash = AmazonBedrockInferenceClient.getModelKeysAndRegionHashcode(model, timeout);
         cacheLock.readLock().lock();
         try {
-            return clientsCache.computeIfAbsent(modelHash, hashKey -> {
-                final AmazonBedrockBaseClient builtClient = creator.apply(model, timeout);
-                builtClient.setClock(clock);
-                builtClient.resetExpiration();
-                return builtClient;
+            return clientsCache.compute(modelHash, (hashKey, client) -> {
+                if (client == null) {
+                    final AmazonBedrockBaseClient builtClient = creator.apply(model, timeout);
+                    builtClient.setClock(clock);
+                    builtClient.resetExpiration();
+                    return builtClient;
+                } else {
+                    // for testing
+                    client.setClock(clock);
```

Comment:
nit: This is completely unrelated to your PR but I wonder if we could refactor the client class' creator function to take the clock as a parameter so we can avoid the `setClock()` class here 🤔

Reply:
I did not tackle this comment in this PR but I left a code comment. Would you like me to open an issue?